.github/workflows/ci-publish.yml

```diff
@@ -16,21 +16,18 @@ jobs:
 
       - name: Checkout code
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
 
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: '3.9'
 
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
-        with:
-          virtualenvs-create: true
-          virtualenvs-in-project: true
-
       - name: Build
         run: |
           python -m pip install --upgrade pip
+          pip install poetry
           pip install poetry-dynamic-versioning
           poetry build
```

Comment:
Minor suggestion, could put "pip install" commands in separate job called "Install Poetry" to keep things organised